rust-app/src/implementation.rs

```diff
@@ -224,7 +232,51 @@ pub static SIGN_IMPL: SignImplT = Action(
     }),
 );
 
+pub type SignHashImplT = impl InterpParser<SignHashParameters, Returning = ArrayVec<u8, 128_usize>>;
+
+pub static SIGN_HASH_IMPL: SignHashImplT = Action(
+    Preaction( || -> Option<()> { write_scroller("Signing", |w| Ok(write!(w, "Transaction Hash")?)) } ,
```

Comment:
@jonored is Preaction subsumed by Bind?

Reply:
I don't think so; Preaction is "I want to inject a side effect before parsing anything". I'm sure we could come up with a way to write something more general, but I don't think it's trivially related to bind. We'd rather not emulate it with something that impacts the schema type, as well.

In this particular case, we don't need it; we don't lose much of anything by having the "Signing Transaction Hash" prompt happen after we've parsed the hash, around line 244. For the full parsed transaction, we're using it so that "Signing Transaction" happens before the other prompts, because the prompts under KadenaCmdInterp can happen in mostly arbitrary order based on how the transaction happened to be serialized.